src/TrackPanel.cpp

```diff
@@ -754,18 +760,36 @@ void TrackPanel::OnMouseEvent(wxMouseEvent & event)
         if(parent)
         {
             wxMenu contextMenu;
-            contextMenu.Bind(wxEVT_MENU,
-               [this]( wxCommandEvent &event ){
-                  OnContextMenu( /*GetProject(),*/ event ); }
-            );
             
-            contextMenu.Append(kContextMenuItemID_AddMonoTrack, XO("Add mono track").Translation());
-            contextMenu.Append(kContextMenuItemID_AddStereoTrack, XO("Add stereo track").Translation());
-            contextMenu.Append(kContextMenuItemID_AddLabelTrack, XO("Add label track").Translation());
-            contextMenu.AppendSeparator();
-            contextMenu.Append(kContextMenuItemID_ExportProject, XO("Export project\tCtrl+E").Translation());
-            contextMenu.AppendSeparator();
-            contextMenu.Append(kContextMenuItemID_SelectAll, XO("Select All\tCtrl+A").Translation());
+            // Get cursor position area: on track or non track (empty area).
+            const auto track = FindTrack (FindCell(event.m_x, event.m_y).pCell.get());
+            if(track)
+            {
+                contextMenu.Append(kContextMenuItemID_Cut, XO("Cut\tCtrl+X").Translation());
+                contextMenu.Append(kContextMenuItemID_Copy, XO("Copy\tCtrl+C").Translation());
```

Comment:
I don't like these repetitious calls to Append.  The description of the menu should be declarative.  There should just be a table of command names, for lookup in CommandManager, and user-visible names.  Use something special like null for the separators.  Then write one simple loop that iterates the table, and make menu items or separators.